harmonica/tests/test_eq_sources_cartesian.py

```diff
@@ -407,3 +405,69 @@ def test_backward_eqlharmonic(depth_type):
         eqs.grid(upward=2e3, shape=shape, region=region),
         eql_harmonic.grid(upward=2e3, shape=shape, region=region),
     )
+
+
+@run_only_with_numba
+@pytest.mark.parametrize("block_size", (None, 500), ids=["no_blocks", "blocks"])
+@pytest.mark.parametrize("custom_points", (False, True), ids=["no_points", "points"])
+@pytest.mark.parametrize("weights_none", (False, True), ids=["no_weights", "weights"])
+@pytest.mark.parametrize("damping", (None, 0.1), ids=["damping_none", "damping"])
+@pytest.mark.parametrize("dtype", ("float64", "float32"))
+def test_dtype(
+    region,
+    coordinates,
+    data,
+    weights,
+    block_size,
+    custom_points,
+    weights_none,
+    damping,
+    dtype,
+):
+    """
+    Test dtype argument on EquivalentSources
+    """
+    # Define the points argument for EquivalentSources
+    points = None
+    if custom_points:
+        points = vd.grid_coordinates(region, spacing=300, extra_coords=-2e3)
+    # Define the points argument for EquivalentSources.fit()
+    if weights_none:
+        weights = None
+    # Initialize and fit the equivalent sources
+    eqs = EquivalentSources(
+        damping=damping, points=points, block_size=block_size, dtype=dtype
+    )
+    eqs.fit(coordinates, data, weights)
+    # Make some predictions
+    prediction = eqs.predict(coordinates)
+    # Check data type of created objects
+    for coord in eqs.points_:
+        assert coord.dtype == np.dtype(dtype)
+    assert prediction.dtype == np.dtype(dtype)
+    # Check the data type of the source coefficients
+    #  assert eqs.coefs_.dtype == np.dtype(dtype)
```

Comment:
@leouieda I've noticed that the `coefs_` are casted to `float64` even if the inputs to `vdb.least_squares` are all `float32`. Even that, the predictions are created as `float32`.

Reply:
I was trying to track this down. It seems that scikit-learn uses the dtype of the Jacobian for everything. But the coeffs are actually coming from [scipy.linalg.solve](https://docs.scipy.org/doc/scipy/reference/generated/scipy.linalg.solve.html#scipy.linalg.solve) according to [this line](https://github.com/scikit-learn/scikit-learn/blob/0d378913be6d7e485b792ea36e9268be31ed52d0/sklearn/linear_model/_ridge.py#L156). The docs for `solve` don't mention dtype so it could be that they always return a `float64` from the underlying LAPACK functions. 

It would be fine to leave the coefs the way they are returned since they usually don't require much storage. It's mostly the Jacobian that's affected.